runtime/compiler/optimizer/FearPointAnalysis.cpp

```diff
@@ -34,10 +34,20 @@
 #include "infra/BitVector.hpp"
 #include "infra/Checklist.hpp"
 
-bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp)
+
+bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp, TR::Node *virtualGuardNode)
    {
-   static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear") == NULL);
-   return kill && !(!comp->getOption(TR_DisableVectorAPIExpansion) && comp->getMethodSymbol()->hasVectorAPI());
+   static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear")) == NULL;
+   if (kill && !comp->getOption(TR_DisableVectorAPIExpansion))
```

Comment:
I think it would be good for this condition to still check `comp->getMethodSymbol()->hasVectorAPI()`, since if that's false, there's no point checking the package name

Reply:
Added that check in https://github.com/eclipse-openj9/openj9/pull/16354/commits/65dee583088499cc7e1787e743d0191096fae72e so now if we do not set environment flag to for all virtual Guards to not kill fear and vector api expansion is not disabled and method contains intrinsic call then it will check the method call that virtual guard is generated for.